cmd/root.go

```diff
@@ -158,68 +181,91 @@ func rootCmd(o *options.Options) error {
 		opts = append(opts, scorecard.WithFileModeGit())
 	}
 
-	repoResult, err = scorecard.Run(ctx, repo, opts...)
-	if err != nil {
-		return fmt.Errorf("scorecard.Run: %w", err)
+	var allResults []*scorecard.Result
+	// Iterate and scan each repo using a helper to keep rootCmd small.
+	for _, uri := range repoURLs {
+		res, err := processRepo(ctx, uri, o, enabledProbes, enabledChecks, checks, opts, checkDocs, pol)
+		if err != nil {
+			// processRepo already logged details; skip this URI.
+			fmt.Fprintf(os.Stderr, "Skipping %s: %v\n", uri, err)
+			continue
+		}
```

Comment:
this changes the behavior of the `scorecard` binary. Previously if any check had a runtime error, `rootCmd` would return an error, which would lead to a non-zero exit code. But now these runtime errors don't lead to an exit code.

I think continuing to scan all repositories is good, but we should "remember" if we saw an error along the way so we can return it at the end

Reply:
Makes sense. I added a small implementation for this using `sawRuntimeErr`. We can improve it, but I think it already aligns with your point.

Reply:
Thanks, I may change the implementation in a follow-up as it would just be more effective for me to handle my suggested changes.